### PR TITLE
Update MemberStatus values to the currently available through the AWS…

### LINKED ISSUE #16403: Securityhub recent MemberStatus changes
AWS released changes today to SecurityHub that changed the MemberStatus fields to contain a few different values then is currently supported by the terraform AWS provider. Because of this change the "invited" field is not correctly calculated when retrieving the current SecurityHub member subscription

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform: v0.12.29
AWS Provider: 3.17.0

### Affected Resource(s)

* aws_securityhub_member

### Terraform Configuration Files

```hcl
resource "aws_securityhub_member" "member" {
  account_id = 00000000000
  email      = my.email@email.com
  invite     = true
}
```

### Debug Output

https://gist.github.com/reactiveinactive/862d8b3e4928918b95c06983e7418f1b
### Expected Behavior

After the Securityhub invite is accepted by the master the terraform run should run without any changes

### Actual Behavior
After accepting the securityhub invitation the resource assumes the invitation is not performed yet and tries to perform a new one.
-/+ resource "aws_securityhub_member" "member" {
        account_id    = "00000000000"
        email         = "my.email@email.com"
      ~ id            = "00000000000" -> (known after apply)
      ~ invite        = false -> true # forces replacement
      ~ master_id     = "11111111111111" -> (known after apply)
      ~ member_status = "Enabled" -> (known after apply)
    }

### Steps to Reproduce
1.  Create a plan with a securityhub member
2.  Run the plan
3.  Accept the invitation
4.  Run the plan again

### Important Factoids
Issue is occurring since today due to recent AWS API changes
https://docs.aws.amazon.com/securityhub/1.0/APIReference/API_Member.html
https://github.com/aws/aws-sdk-java/blame/a23c826512114220ff623b9d449489c3822b058e/aws-java-sdk-securityhub/src/main/java/com/amazonaws/services/securityhub/model/Member.java#L71

### PR DESCRIPTION
**Pull request by reactiveinactive**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #16403

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/aws_securityhub_member: Update MemberStatus values 
```

Output from acceptance testing:
```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by bflad** [comment]
Please note that I'm holding off on merging this as-is, since the service team may opt to rollback the change since it breaks all previous Terraform AWS Provider versions. We could temporarily support both the previous and new value if we wanted to get this in immediately without risk of the API changing again.

**Comment by bflad** [comment]
(In the meantime, for anyone 👍  reacting here, you may also want to 👍  react to the issue itself: https://github.com/hashicorp/terraform-provider-aws/issues/16403 so there's accurate tracking of those affected)

File: aws/resource_aws_securityhub_member.go
Comment (bflad) [inline aws/resource_aws_securityhub_member.go:18]:
```suggestion
	// Associated is the member status naming for regions that do not support Organizations
	SecurityHubMemberStatusAssociated = "Associated"
	SecurityHubMemberStatusCreated    = "Created"
	SecurityHubMemberStatusInvited    = "Invited"
	SecurityHubMemberStatusEnabled    = "Enabled"
	SecurityHubMemberStatusRemoved    = "Removed"
	SecurityHubMemberStatusResigned   = "Resigned"
	SecurityHubMemberStatusDeleted    = "Deleted"
```

File: aws/resource_aws_securityhub_member.go
Comment (bflad) [inline aws/resource_aws_securityhub_member.go:132]:
```suggestion
	invited := status == SecurityHubMemberStatusInvited || status == SecurityHubMemberStatusEnabled || status == SecurityHubMemberStatusAssociated || status == SecurityHubMemberStatusResigned
```

**Review by bflad — CHANGES_REQUESTED** [review]
Thank you for raising this, @reactiveinactive

Since AWS China (and potentially other AWS partitions) do not support the new Organizations integration that caused this breaking API change:

* https://aws.amazon.com/about-aws/whats-new/2020/11/aws-security-hub-integrates-with-aws-organizations-for-simplified-security-posture-management/
* https://forums.aws.amazon.com/ann.jspa?annID=8164)
* https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-regions.html

We will need to support both old and new values for now. Once this is in, we should be able to  merge this and release it in the next day or two.

**Comment by reactiveinactive** [comment]
@bflad looks ok to me, do I need to do anything else? First time doing a pull request so not entirely sure on how this all works.

**Review by bflad — APPROVED** [review]
Looks good, thank you for this fix, @reactiveinactive 🚀 

Output from acceptance testing:

```
--- PASS: TestAccAWSSecurityHub_serial (29.45s)
    --- PASS: TestAccAWSSecurityHub_serial/Member (29.45s)
        --- PASS: TestAccAWSSecurityHub_serial/Member/basic (15.06s)
        --- PASS: TestAccAWSSecurityHub_serial/Member/invite (14.39s)
```

**Pull request merged**
